extensions/vertx-web/deployment/src/test/java/io/quarkus/vertx/web/VertxWebDevModeTestCase.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.vertx.web;
+
+import java.util.function.Function;
+
+import org.hamcrest.Matchers;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusDevModeTest;
+import io.restassured.RestAssured;
+
+public class VertxWebDevModeTestCase {
+
+    @RegisterExtension
+    static QuarkusDevModeTest runner = new QuarkusDevModeTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(TestRoute.class));
+
+    @Test
+    public void testRunningInDevMode() {
+        RestAssured.given()
+                .body("OK")
+                .post("/test")
+                .then().statusCode(200)
+                .body(Matchers.equalTo("test route"));
+
+        runner.modifySourceFile(TestRoute.class, new Function<String, String>() {
```

Comment:
How about updating `TestRoute` so that the handler calls executing blocking, and we verify that the context when calling `executingBlocking` and when the callback is involved is the same:

```java
Vertx vertx = rc.vertx()
expected = vertx.getOrCreateContext();
vertx.executeBlocking(Promise::complete, ar -> {
  actual = vertx.getOrCreateContext();
  // expected should be the same as actual
  // before AND after hot reload
})
```

Reply:
I have updated the test to verify the connection vertx instance matches the current vertx instance.